mkdocs/docs/protocol/onchain-credential-status.md

```diff
@@ -0,0 +1,200 @@
+# OnChain Credential Status
+
+On-chain identity verification requires the ability to check whether a claim has been revoked using a smart contract. To accomplish this, the claim must include information about the smart contract, such as its address, blockchain, and network. This information can be stored in the `CredentialStatus` structure.
+
+The `CredentialStatus` structure should contain the following fields:
+
+```golang
+type CredentialStatus struct {
+	ID                 string               `json:"id"`
+	Type               CredentialStatusType `json:"type"`
+	RevocationNonce    uint64               `json:"revocationNonce"`
+}
+```
+
+The `ID` field is a composite field that contains encoded information in the following format:
+
+`did:[did-method]:[blockchain]:[network]:[id]/credentialStatus?(revocationNonce=value)&(contractAddress={chainID}:{contractAddress})`
+
+`type` - credential type
+
+`revocationNonce` - unique revocation nonce
+
+Example:
+
+```json
+{
+"id": "did:polygonid:polygon:main:2qCU58EJgrEMWhziKqC3qNXJkZPY8XCxDSBM4mqPkM/credentialStatus?revocationNonce=1234&contractAddress=1:0xf3bB959314B5D1e4587e1f597ccc289216608ac5",
+"type": "Iden3OnchainSparseMerkleTreeProof2023",
+"revocationNonce": "1234"
+}
+```
+
+The `revocationNonce` and `contractAddress` parameters inside the ID field are optional. Here's an example without them:
+
+```json
+{
+"id": "did:polygonid:polygon:main:2qCU58EJgrEMWhziKqC3qNXJkZPY8XCxDSBM4mqPkM/credentialStatus",
+"type": "Iden3OnchainSparseMerkleTreeProof2023",
+"revocationNonce": "1234"
+}
+```
+
+## Process ID field
+
+In the context of `OnChainCredentialStatus`, the `id` field can contain two additional optional parameters: `revocationNonce` and `contractAddress`.
+`revocationNonce` is a credential revocation `nonce`, while `contractAddress` is the address of a smart contract that implements an on-chain issuer interface. The `contractAddress` field is composed by two parts: `chainID` and `contractAddress`. Use `chainID` to select the correct network.
+
+If `contractAddress` is not set, find the default contract address by parsing DID [extracting the on-chain issuer contract address](https://github.com/iden3/go-iden3-core/blob/014f51e92da5c0c89c95c31e42bfca1652d2ad14/did.go#L345-L354) and getting `chainID` from the DID network. Use the blockchain name, network and contract address from the DID to make on-chain revocation request. If the DID doesn't have a contract address inside and `contractAddress` parameter is empty, this VC document should be considered invalid.
+
+If `revocationNonce` is not set, the `revocationNonce` value from the struct will be used instead.
+
+### Extract core.Id from the Credential Status ID field
```

Comment:
Move these two steps to ##Worflow section, please.

Reply:
In my opinion, we can simplify these two steps in the documentation to:
1. Parse core.DID from credentialStatus.id filed with [js](https://github.com/iden3/js-iden3-core/blob/baa0ead8a3e2340bb4d78132ec63e6e24d806da9/src/did/did.ts#L160)|[go](https://github.com/iden3/go-iden3-core/blob/014f51e92da5c0c89c95c31e42bfca1652d2ad14/w3c/did_w3c.go#L165)
2. Extract core.Id fome core.DID with [js](https://github.com/iden3/js-iden3-core/blob/baa0ead8a3e2340bb4d78132ec63e6e24d806da9/src/did/did.ts#L160)|[go](https://github.com/iden3/go-iden3-core/blob/014f51e92da5c0c89c95c31e42bfca1652d2ad14/did.go#L184)